### PR TITLE
Add basic pdf reader

### PR DESCRIPTION
**Pull request by dkanada**

### CODE REVIEW

**Comment by thornbill** [comment]
Lint is not happy:
```
/home/vsts/work/1/s/src/plugins/pdfPlayer/plugin.js
   31:13  error  'elem' is never reassigned. Use 'const' instead          prefer-const
   38:13  error  'elem' is never reassigned. Use 'const' instead          prefer-const
   80:13  error  'key' is never reassigned. Use 'const' instead           prefer-const
  114:13  error  'elem' is never reassigned. Use 'const' instead          prefer-const
  128:13  error  'elem' is never reassigned. Use 'const' instead          prefer-const
  177:13  error  'item' is never reassigned. Use 'const' instead          prefer-const
  188:13  error  'serverId' is never reassigned. Use 'const' instead      prefer-const
  189:13  error  'apiClient' is never reassigned. Use 'const' instead     prefer-const
  193:21  error  'downloadHref' is never reassigned. Use 'const' instead  prefer-const
  198:21  error  'downloadTask' is never reassigned. Use 'const' instead  prefer-const
  242:13  error  'pages' is never reassigned. Use 'const' instead         prefer-const
  249:18  error  'page' is never reassigned. Use 'const' instead          prefer-const
  260:18  error  'page' is never reassigned. Use 'const' instead          prefer-const
  281:13  error  Unexpected var, use let or const instead                 no-var
  286:17  error  'renderTask' is never reassigned. Use 'const' instead    prefer-const
```

**Review by YouKnowBlom — APPROVED** [review]
Untested but the code lgtm

File: src/plugins/bookPlayer/plugin.js [hunk 1]
```diff
@@ -271,40 +268,32 @@ export class BookPlayer {
...
                     epubElem.style.display = 'none';
 
                     this.bindEvents();
 
-                    return this._rendition.book.locations.generate(1024).then(async () => {
-                        if (cancellationToken.shouldCancel) {
-                            return reject();
-                        }
+                    return this.rendition.book.locations.generate(1024).then(async () => {
+                        if (this.cancellationToken) reject();
```
Comment (oddstr13) [inline src/plugins/bookPlayer/plugin.js:281]: I'm not a fan of brace-less if statements, but I don't know how common a pattern that is in our code-base.

File: src/plugins/bookPlayer/plugin.js [hunk 2]
```diff
@@ -320,7 +309,7 @@ export class BookPlayer {
     }
 
     canPlayItem(item) {
-        if (item.Path && (item.Path.endsWith('epub'))) {
+        if (item.Path && item.Path.endsWith('epub')) {
```
Comment (oddstr13) [inline src/plugins/bookPlayer/plugin.js:312]: Is `item.Path` file ending always lowercase?

File: src/plugins/pdfPlayer/plugin.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+            });
+        });
+    }
+
+    canPlayMediaType(mediaType) {
+        return (mediaType || '').toLowerCase() === 'book';
+    }
+
+    canPlayItem(item) {
+        if (item.Path && item.Path.endsWith('pdf')) {
```
Comment (oddstr13) [inline src/plugins/pdfPlayer/plugin.js:298]: Is `item.Path` file ending always lowercase?

File: src/plugins/bookPlayer/plugin.js (hunk 1 above)
Comment (dkanada) [reply src/plugins/bookPlayer/plugin.js:281]: It's relatively common in this codebase. We can always add a rule later I suppose, but this repository is getting less active lately.

File: src/plugins/bookPlayer/plugin.js (hunk 2 above)
Comment (dkanada) [reply src/plugins/bookPlayer/plugin.js:312]: I believe it should be, I have yet to come accross this issue.

Comment (oddstr13) [reply src/plugins/bookPlayer/plugin.js:312]: Well, I guess I know what I'll be naming some of my ebook files then. :sweat_smile:

**Review by oddstr13 — APPROVED** [review]

File: src/plugins/bookPlayer/plugin.js (hunk 2 above)
Comment (oddstr13) [reply src/plugins/bookPlayer/plugin.js:312]: This match is indeed case sensitive, and won't recognize anything but lowercase `epub` (and `pdf`) as "playable".

**Pull request merged**